src/Program.cs

```diff
@@ -21,10 +19,14 @@ public static void Main(string[] args)
             Host.CreateDefaultBuilder(args)
                 .ConfigureSecretStore((context, config, builder) =>
                 {
-                    builder.AddProvider(new InMemorySecretProvider(new Dictionary<string, Secret>
-                    {
-                        ["MySecret"] = new Secret("123", "122asad-AD3-SDAF3223")
-                    }));
+                    builder.AddEnvironmentVariableProvider()
+                           .AddAzureKeyVaultProvider(
```

Comment:
We might want to make this easier meaning if you provide:
- "client-id", "client-key" we use ServicePrincipalAuthentication for you after the scenes
- nothing MSI for you after the scenes
- vault URL will be wrapped in KeyVaultConfiguration

So that it's easier for folks to onboard without having to care

Reply:
Ah, yes, and maybe with with different overloads then, But maybe with a 'expert overload' as well, that takes in the same values as the `KeyVaultSecretProvider` for custom implementation.

Reply:
That's fine for me!